src/multiversioning.zig

```diff
@@ -1119,15 +1194,49 @@ pub const Multiversion = struct {
         self.stage = .ready;
     }
 
-    fn handle_error(self: *Multiversion, result: anyerror) void {
+    fn handle_error(self: *MultiversionOS, result: anyerror) void {
         assert(self.stage != .init);
 
         log.err("binary does not contain valid multiversion data: {}", .{result});
 
         self.stage = .{ .err = result };
     }
 
-    pub fn exec_current(self: *Multiversion, release_target: Release) !noreturn {
+    fn replica_release_execute(self: *MultiversionOS, release: Release) noreturn {
+        assert(release.value != constants.config.process.release.value);
+        assert(release.value != Release.zero.value);
+        assert(release.value != Release.minimum.value);
+
+        if (!self.releases_bundled.contains(release)) {
+            log.err("release_execute: release {} is not available;" ++
+                " upgrade (or downgrade) the binary", .{
+                release,
+            });
+            @panic("release not available");
+        }
+
+        // We have two paths here, depending on if we're upgrading or downgrading. If we're downgrading
+        // the invariant is that this code is running _before_ we've finished opening, that is,
+        // release_transition is called in open().
+        if (release.value < constants.config.process.release.value) {
```

Comment:
This used to be 

```suggestion
        if (release.value < replica.release.value) {
```

but replica.realease is config.process.release